bsb/morphologies/__init__.py

```diff
@@ -1132,7 +1132,11 @@ def versor(self):
         Return the normalized vector of the axis connecting the start and terminal points.
         """
         try:
-            return (self.end - self.start) / np.linalg.norm(self.end - self.start)
+            versor = (self.end - self.start) / np.linalg.norm(self.end - self.start)
+            if np.isnan(versor).any():
```

Comment:
This is an inefficient check, as for branches that work perfectly fine, of N points, N non-NaN points would be checked for NaN-ness.

Reply:
Ok, if versor is always a single number:

```suggestion
            versor = (self.end - self.start) / np.linalg.norm(self.end - self.start)[0]
            if np.isnan(versor):
```

Reply:
Ok, didn't know! Then see suggestion below